src/fam-api/fam_ops_libfabric.cpp

```diff
@@ -374,32 +372,1134 @@ void Fam_Ops_Libfabric::finalize() {
 }
 
 int Fam_Ops_Libfabric::put_blocking(void *local, Fam_Descriptor *descriptor,
-                                    uint64_t offset, uint64_t nbytes) {
-    std::ostringstream message;
-    // Write data into memory region with this key
-    uint64_t key;
-    key = descriptor->get_key();
-    offset += (uint64_t)descriptor->get_base_address();
-    uint64_t nodeId = descriptor->get_memserver_id();
-    std::vector<fi_addr_t> *fiAddr = get_fiAddrs();
-    int ret = fabric_write(key, local, nbytes, offset, (*fiAddr)[nodeId],
-                           get_context(descriptor));
-    return ret;
+		                                    uint64_t offset, uint64_t nbytes) {
+	uint64_t *memServerIds = descriptor->get_memserver_ids();
+    size_t interleaveSize = descriptor->get_interleave_size();
+    uint64_t *keys = descriptor->get_keys();
+    void **base_addr_list = descriptor->get_base_address_list();
+    uint64_t usedMemsrvCnt = descriptor->get_used_memsrv_cnt();
+    uint64_t startIndex = ((offset/interleaveSize)%usedMemsrvCnt);
+    uint64_t startOffset = ((offset/interleaveSize) - startIndex)/usedMemsrvCnt;
```

Comment:
Do you think renaming this variable will work better - to something like startBlock or currentBlock.